storage/engine/mvcc.go

```diff
@@ -497,7 +498,7 @@ func mvccGetInternal(engine Engine, key proto.Key, metaKey proto.EncodedKey, tim
 		if meta.Txn != nil && (txn == nil || !bytes.Equal(meta.Txn.ID, txn.ID)) {
```

Comment:
this part could be untangled a little bit now that we're also possibly returning the same error in other cases. But let's leave it for this PR.